### PR TITLE
[PREVIEW] Send envelope message

### PR DESCRIPTION
**Pull request by luigibk**
### JIRA link (if applicable) ###
RPE-648

### Change description ###
Send envelope message to service bus for consumption by orchestrator.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java [hunk 1]
```diff
@@ -10,4 +10,5 @@
     DOC_CONSUMED, // client service handled the documents
     BLOB_DELETE_FAILURE, // when blob is not successfully deleted after processing is complete
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java]: not related to this PR, but is this still needed/used?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java [hunk 2]
```diff
@@ -33,18 +37,27 @@ protected Processor(
...
             return;
         }
-        markAsProcessed(envelope);
+        if (!markAsProcessed(envelope)) {
+            return;
+        }
+        if (!sendProcessedMessage(serviceBusHelper, envelope)) {
+            return;
+        }
+        markAsNotified(envelope);
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java]: `markAsProcessed` is wrapped, this is not.
Is this by design?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java (hunk 1 above)
Comment (luigibk) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java]: That's an event not an envelope status. It is still used as we might want to track how many times blob delete fails.

Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/entity/Event.java]: ok

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java (hunk 2 above)
Comment (luigibk) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java]: They are all wrapped by handleEvent() for what I can see.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestWithAcquireLease.java [hunk 3]
```diff
@@ -78,7 +78,7 @@ public void should_process_zip_file_only_once_by_processor_thread_which_acquires
         // and
         // We expect only two events one for doc upload and another for doc processed
         List<ProcessEvent> processEvents = processEventRepository.findAll();
-        assertThat(processEvents).hasSize(2);
+        assertThat(processEvents).hasSize(3);
```
Comment (rkondratowicz) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestWithAcquireLease.java]: Comment in line 80 needs updating.
Or consider updating this test to check the items, not just the count.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java [hunk 4]
```diff
@@ -53,6 +55,16 @@ public BlobProcessorTask(
...
 
+    /**
+     * Spring overrides the {@code @Lookup} method and returns an instance of bean.
+     *
+     * @return Instance of {@code ServiceBusHelper}
+     */
+    @Lookup
+    public ServiceBusHelper serviceBusHelper() {
+        return null;
+    }
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: can we inject this dependency the same way as all the other ones? (via constructor)

Comment (luigibk) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: This way a new service bus client is created for every run of the task. If we inject it like the other dependencies there will only be a single client for the entire lifetime of the application. For the time being I think this is safer as the client gets renewed.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java (hunk 2 above)
Comment (luigibk) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/Processor.java]: It is definitely wrapped.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/DocProcessedNotificationFailureException.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    private final transient Envelope envelope;
+
+    public DocProcessedNotificationFailureException(Envelope envelope) {
+        super("Message send failure");
+
+        this.envelope = envelope;
+    }
+
+    public DocProcessedNotificationFailureException(Envelope envelope, Throwable cause) {
+        super(cause.getMessage(), cause);
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/DocProcessedNotificationFailureException.java]: I'd probably set a fixed message here, but don't feel strong about it.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java (hunk 4 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java]: I agree - used to have a problem with the client not recovering after a broken connection.

**Pull request merged**